test/tools_tests/branch_length_estimator_tests/iid_exponential_mle_test.py

```diff
@@ -359,3 +360,244 @@ def test_minimum_branch_length(self, name, solver):
+        Prior to rescaling, the first branch is of length
+        Ln[(rate_1+rate_2+rate_3)/(rate_2+rate_3)]/rate_1 and the other is
+        of length equal to Ln[(rate_2+rate_3)/rate_3]/rate_2.
+        """
+        rate_1, rate_2, rate_3 = solver_rates
+        math_rate_1, math_rate_2, math_rate_3 = math_rates
+
+        tree = nx.DiGraph()
+        tree.add_nodes_from(["0", "1", "2"])
+        tree.add_edge("0", "1")
+        tree.add_edge("1", "2")
+        tree = CassiopeiaTree(tree=tree)
+        tree.set_all_character_states(
+            {"0": [0, 0, 0], "1": [1, 0, 0], "2": [1, 1, 0]}
+        )
+        relative_rates = [rate_1, rate_2, rate_3]
+        model = IIDExponentialMLE(
+            minimum_branch_length=1e-4, relative_mutation_rates=relative_rates
+        )
+        model.estimate_branch_lengths(tree)
+
+        branch1 = (
+            math.log(
+                (math_rate_1 + math_rate_2 + math_rate_3)
+                / (math_rate_2 + math_rate_3)
+            )
+            / math_rate_1
+        )
+        branch2 = (
+            math.log((math_rate_2 + math_rate_3) / math_rate_3) / math_rate_2
+        )
+        total = branch1 + branch2
+        branch1, branch2 = branch1 / total, branch2 / total
+        mutation_rates = [x * total for x in relative_rates]
+
+        should_be_equal = True
+        for r1, r2 in zip(solver_rates, math_rates):
+            if r1 != r2:
+                should_be_equal = False
+                break
+
+        if should_be_equal:
+            self.assertAlmostEqual(
+                tree.get_branch_length("0", "1"), branch1, places=3
+            )
+            self.assertAlmostEqual(
+                tree.get_branch_length("1", "2"), branch2, places=3
+            )
+            self.assertAlmostEqual(tree.get_time("0"), 0.0, places=3)
+            self.assertAlmostEqual(tree.get_time("1"), branch1, places=3)
+            self.assertAlmostEqual(tree.get_time("2"), 1.0, places=3)
+            for x, y in zip(model.mutation_rate, mutation_rates):
+                self.assertAlmostEqual(x, y, places=3)
+        else:
+            with self.assertRaises(AssertionError):
+                self.assertAlmostEqual(
+                    tree.get_branch_length("0", "1"), branch1, places=3
+                )
+            with self.assertRaises(AssertionError):
+                self.assertAlmostEqual(
+                    tree.get_branch_length("1", "2"), branch2, places=3
+                )
+            self.assertAlmostEqual(tree.get_time("0"), 0.0, places=3)
+            with self.assertRaises(AssertionError):
+                self.assertAlmostEqual(tree.get_time("1"), branch1, places=3)
+            self.assertAlmostEqual(tree.get_time("2"), 1.0, places=3)
+            for x, y in zip(model.mutation_rate, mutation_rates):
+                with self.assertRaises(AssertionError):
+                    self.assertAlmostEqual(x, y, places=3)
+
+    def test_invalid_site_rates(self):
```

Comment:
Essentially for free (using @parameterized to try different `relative_rates`) we can cover more cases here, such as `relative_rates = [1.5, 0, 2.5]` (zero element), `relative_rates = [1, 1]` (too few rates) and `relative_rates = [1, 1, 1, 1]` (too many rates).